themes/palenight-italic.json

```diff
@@ -1181,53 +1181,95 @@
       }
     },
     {
-      "name": "Markdown Headings",
-      "scope": "markup.heading.markdown",
+      "name": "Markup Headings",
+      "scope": "markup.heading",
       "settings": {
         "foreground": "#82b1ff"
       }
     },
     {
-      "name": "Markdown Italics",
-      "scope": "markup.italic.markdown",
+      "name": "Markup Italics",
+      "scope": "markup.italic",
       "settings": {
         "foreground": "#c792ea",
         "fontStyle": "italic"
       }
     },
     {
-      "name": "Markdown Bold",
-      "scope": "markup.bold.markdown",
+      "name": "Markup Bold",
+      "scope": "markup.bold",
       "settings": {
         "foreground": "#ffcb6b",
         "fontStyle": "bold"
       }
     },
     {
-      "name": "Markdown Quote + others",
-      "scope": "markup.quote.markdown",
+      "name": "Markup Quote + others",
+      "scope": "markup.quote",
       "settings": {
         "foreground": "#697098",
         "fontStyle": "italic"
       }
     },
     {
-      "name": "Markdown Raw Code + others",
-      "scope": "markup.inline.raw.markdown",
+      "name": "Markup Raw Code + others",
+      "scope": "markup.inline.raw",
       "settings": {
         "foreground": "#80CBC4"
       }
     },
     {
-      "name": "Markdown Links",
+      "name": "Markup Links",
       "scope": [
-        "markup.underline.link.markdown",
-        "markup.underline.link.image.markdown"
+        "markup.underline.link",
+        "markup.underline.link.image"
       ],
       "settings": {
         "foreground": "#ff869a"
       }
     },
+    {
+      "name": "Markup Fenced Code",
+      "scope": [
+        "fenced_code.block.language"
```

Comment:
Do we need the `language` identifier? Would `fenced_code.block` alone work?

Reply:
I thought it was necessary in order to not paint the whole content in green, but it seems you are totally right!
The language name is prefixed by `fenced_code.block`, but the content itself is prefixed by `markup.fenced_code.block`.
This discrepancy is surprising.

In this snippet:
~~~markdown
```
Sample text here...
```

``` js
var foo = function (bar) {
  return bar++;
};
console.log(foo(5));
```
~~~
Both `fenced_code.block` and `fenced_code.block.language` renders like this:
![image](https://user-images.githubusercontent.com/763200/93851476-7d02ca80-fcb0-11ea-9965-0a37d319454c.png)

If we change it to `markup.fenced_code.block`, unknown or unspecified language are colored in green (which is nice I think) but it also impacts the inner code highlighting for known languages. Like the curly braces and semicolons in this JS code:
![image](https://user-images.githubusercontent.com/763200/93851540-9c99f300-fcb0-11ea-92e1-49b029af0f7a.png)

Reply:
Oh, maybe this would be better?
~~~js
          {
            "name": "Markup Fenced Code",
            "scope": [
              "markup.fenced_code.block"
            ],
            "settings": {
              "foreground": "#80CBC4",
            }
          },
          {
            "name": "Markup Fenced Code Inner code",
            "scope": [
              "markup.fenced_code.block meta.embedded.block"
            ],
            "settings": {
              "foreground": "#BFC7D5",
            }
          },
~~~
It renders like this:
![image](https://user-images.githubusercontent.com/763200/93852371-34e4a780-fcb2-11ea-92d9-78e84a39cc55.png)